lib/discourse.rb

```diff
@@ -814,7 +814,7 @@ def self.deprecate(warning, drop_from: nil, since: nil, raise_error: false, outp
     digest = Digest::MD5.hexdigest(warning)
     redis_key = "deprecate-notice-#{digest}"
 
-    if !Discourse.redis.without_namespace.get(redis_key)
+    if Discourse.redis && !Discourse.redis.without_namespace.get(redis_key)
```

Comment:
Curious why is this change required?

Reply:
I am calling `deprecate` method in `redis_config` which is called before global Redis for Discourse is set. So at this state Discourse.redis is nil

Reply:
It looks like if `Discourse.redis` is `nil`, `Rails.logger.warn(warning)` is not called which means that deprecation message won't be visible. Should we split that out from this check?